### DISCUSSION TITLE
a performance of obtaining token

### DISCUSSION THREAD

**Discussion by sanyaviper** [ory/hydra#3015] [Q&A] [answered]
Hi people,
I'm curious how fast should be Refresh Token Rotation(obtaining a token by a refresh token).
I'm currently using Hydra with PostgreSQL and
my env is:
4 pods of hydra in a cluster that takes less than 30% of CPU usage all the time,
DB (General Purpose, D16s_v3, 16 vCores, 64 GiB RAM, 1048576 storage)
I did load testing and got the next results.
1) 50 parallel requests - gives me ~75 requests/seconds
1) 100 parallel requests - gives me ~70 requests/seconds
2) 200 parallel requests - gives me ~60 requests/seconds (load to DB is getting higher.. )
 
In my opinion, it's too slow but maybe I'm wrong what do you think?
Any suggestions or ideas are welcome.

**Comment by aeneasr** [comment, marked as answer]
The problem here is most likely BCrypt, please see: https://www.ory.sh/docs/hydra/benchmark#bcrypt

We are planning to introduce more hashing methods, including ones that are less secure but also much faster to improve this situation :)
